sql-to-dbsp-compiler/lib/sqllib/src/aggregates.rs

```diff
@@ -3,13 +3,109 @@
 
 use crate::binary::ByteArray;
 use crate::timestamp::*;
+use crate::{FromInteger, ToInteger};
 use core::ops::Add;
-use dbsp::algebra::{F32, F64};
+use dbsp::algebra::{FirstLargeValue, HasOne, HasZero, IsSigned, IsUnsigned, F32, F64};
 use num::PrimInt;
 use rust_decimal::Decimal;
 use std::cmp::Ord;
+use std::fmt::Debug;
 use std::marker::Copy;
 
+/// Holds some methods for wrapping values into unsigned values
+/// This is used by partitioned_rolling_aggregate
+pub struct UnsignedWrapper {}
+
+/// The conversion involves four types:
+/// O: the original type which is being converted
+/// S: a signed type that the original type can be converted to losslessly
+/// I: an intermediate signed type wider than S
+/// U: an unsigned type the same width as I
+//  None of the 'unwrap' calls below should ever fail.
+impl UnsignedWrapper {
+    // Conversion from O to U
+    // O cannot be nullable, so nullsLast is irrelevant
```

Comment:
then why not just remove the argument here and in `to_signed` below?

Reply:
simpler calling convention from Java, always the same 4 arguments.